File: Java/Medium/CompileError/path-with-maximum-probability_7.java
class Solution {
    public double maxProbability(int n, int[][] edges, double[] succProb, int start, int end) {
        
        // Build the graph
        List<int[]>[] graph = new List[n];
        for (int i = 0; i < n; i++) {
            graph[i] = new ArrayList<int[]>();
        }
        for (int i = 0; i < edges.length; i++) {
            graph[edges[i][0]].add(new int[]{edges[i][1], i});
            graph[edges[i][1]].add(new int[]{edges[i][0], i});
        }
        // Dijkstra
        PriorityQueue<int[]> pq = new PriorityQueue<int[]>((a, b) -> Double.compare(b[1], a[1]));
        pq.offer(new int[]{start, 1.0});
        double[] prob = new double[n];
        prob[start] = 1.0;
        while (!pq.isEmpty()) {
            int[] p = pq.poll();
            int u = p[0];
            double w = p[1];
            if (u == end) {
                return w;
            }
            for (int[] v : graph[u]) {
                int next = v[0];
                double nextProb = w * succProb[v[1]];
                if (nextProb > prob[next]) {
                    prob[next] = nextProb;
                    pq.offer(new int[]{next, nextProb});
                }
            }
        }
        return 0.0;
    }
}

File: C++/Medium/Accepted/path-with-maximum-probability_7.cpp
class Solution {
public:
    double maxProbability(int n, vector<vector<int>>& edges, vector<double>& succProb, int start, int end) {
        
        vector<vector<pair<int, double>>> graph(n);
        for (int i = 0; i < edges.size(); ++i) {
            graph[edges[i][0]].push_back(make_pair(edges[i][1], succProb[i]));
            graph[edges[i][1]].push_back(make_pair(edges[i][0], succProb[i]));
        }
        vector<double> prob(n, 0);
        prob[start] = 1;
        priority_queue<pair<double, int>> pq;
        pq.push(make_pair(1, start));
        while (!pq.empty()) {
            auto p = pq.top();
            pq.pop();
            double p1 = p.first;
            int n1 = p.second;
            if (prob[n1] > p1) continue;
            for (auto p2 : graph[n1]) {
                int n2 = p2.first;
                double p3 = p2.second;
                if (prob[n2] < prob[n1] * p3) {
                    prob[n2] = prob[n1] * p3;
                    pq.push(make_pair(prob[n2], n2));
                }
            }
        }
        return prob[end];
    }
};